charts/kserve-resources/templates/configmap.yaml

```diff
@@ -199,9 +198,15 @@ data:
            # NOTE: This configuration only applicable for serverless deployment with Istio configured as network layer.
            "ingressGateway" : "knative-serving/knative-ingress-gateway",
 
-           # ingressService specifies the hostname of the ingress service.
+           # knativeLocalGatewayService specifies the hostname of the Knative's local gateway serivice.
+           # The default KServe configurations are re-using the Istio local gateways for Knative. In this case, this
+           # knativeLocalGatewayService field can be left unset. When unset, the value of "localGatewayService" will be used.
+           # However, sometimes it may be better to have local gateways specifically for KServe (e.g. when enabling strict mTLS in Istio).
+           # Under such setups where KServe is needed to have its own local gateways, the values of the "localGateway" and
+           # "localGatewayService" should point to the KServe local gateways. Then, this knativeLocalGatewayService field
+           # should point to the Knative's local gateway service.
            # NOTE: This configuration only applicable for serverless deployment with Istio configured as network layer.
-           "ingressService" : "istio-ingressgateway.istio-system.svc.cluster.local",
+           "knativeLocalGatewayService": "",
```

Comment:
I can do it in this part, as this is just an example, to be explicit about what would be the default value.

Although, I'm not sure if it is OK to be explicit in the actual configs. I'm not sure if that would lead to configuration mistakes for users that would be migrating and using a different name for the gateways.

Reply:
I am just thinking about a better name for this as by default other gateway configurations are reusing knative gateways, so this could lead to some confusions.

Reply:
The way I think it is that, by default, _kserve gateway_ is also the Knative's one :slightly_smiling_face: .

I was unable to think a better name, given in the de-coupled configuration, this name seems to represent its goal. But if there is a better naming, I'm happy to change it.